presto-geospatial-toolkit/src/test/java/com/facebook/presto/geospatial/TestGeometrySerialization.java

```diff
@@ -0,0 +1,204 @@
+        assertEquals(deserializeEnvelope(geometryFromText("MULTILINESTRING ((1 1, 5 1), (2 4, 4 4))")), new Envelope(1, 1, 5, 4));
+        assertEquals(deserializeEnvelope(geometryFromText("POLYGON ((0 0, 0 4, 4 0))")), new Envelope(0, 0, 4, 4));
+        assertEquals(deserializeEnvelope(geometryFromText("MULTIPOLYGON (((0 0 , 0 2, 2 2, 2 0)), ((2 2, 2 4, 4 4, 4 2)))")), new Envelope(0, 0, 4, 4));
+        assertEquals(deserializeEnvelope(geometryFromText("GEOMETRYCOLLECTION (POINT (3 7), LINESTRING (4 6, 7 10))")), new Envelope(3, 6, 7, 10));
+        assertEquals(deserializeEnvelope(geometryFromText("POLYGON EMPTY")), null);
+        assertEquals(deserializeEnvelope(geometryFromText("POINT (1 2)")), new Envelope(1, 2, 1, 2));
+        assertEquals(deserializeEnvelope(geometryFromText("POINT EMPTY")), null);
+        assertEquals(deserializeEnvelope(geometryFromText("GEOMETRYCOLLECTION (GEOMETRYCOLLECTION (POINT (2 7), LINESTRING (4 6, 7 10)), POINT (3 7), LINESTRING (4 6, 7 10))")), new Envelope(2, 6, 7, 10));
+    }
+
+    private static void testSerialization(String wkt)
+    {
+        testEsriSerialization(wkt);
+        testJtsSerialization(wkt);
+    }
+
+    private static void testEsriSerialization(String wkt)
+    {
+        OGCGeometry expected = OGCGeometry.fromText(wkt);
+        OGCGeometry actual = deserialize(serialize(expected));
+        assertGeometryEquals(actual, expected);
+    }
+
+    private static void testJtsSerialization(String wkt)
+    {
+        Geometry jtsGeometry = createJtsGeometry(wkt);
+        OGCGeometry esriGeometry = OGCGeometry.fromText(wkt);
+
+        Slice jtsSerialized = JtsGeometrySerde.serialize(jtsGeometry);
+        Slice esriSerialized = GeometrySerde.serialize(esriGeometry);
+        assertEquals(jtsSerialized, esriSerialized);
+
+        Geometry jtsDeserialized = JtsGeometrySerde.deserialize(jtsSerialized);
+        assertGeometryEquals(jtsDeserialized, jtsGeometry);
+
+        OGCGeometry esriDeserialized = GeometrySerde.deserialize(esriSerialized);
+        assertGeometryEquals(esriDeserialized, esriGeometry);
+    }
+
+    private static Slice geometryFromText(String wkt)
+    {
+        return serialize(OGCGeometry.fromText(wkt));
+    }
+
+    private static Geometry createJtsGeometry(String wkt)
+    {
+        try {
+            return new WKTReader().read(wkt);
+        }
+        catch (ParseException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void assertGeometryEquals(Geometry actual, Geometry expected)
+    {
+        if (expected.isEmpty()) {
+            assertTrue(actual.isEmpty());
+        }
+        else {
+            assertEquals(actual.norm(), expected.norm());
+        }
+    }
+
+    private static void assertGeometryEquals(OGCGeometry actual, OGCGeometry expected)
+    {
+        if (expected.isEmpty()) {
```

Comment:
Since earlier commits enhanced serialization to avoid losing multi-types this `if` can be removed.

Reply:
It uncovered two  more bugs in `esri` libraries. 

First that it may create multitype instead of a singular type for empty geometry (i had to copy `createFromEsriGeometry` method to our codebase and fix it. Luckily it was fairly simple, and the method itself is simple)

Second bug is a bug in equality comparison. It may return false for two empty linestrings if one of those doesn't have envelope loaded. Had to introduce workaround by forcing envelope loading before equity comparison.